src/data_provider/src/packages/pkgWrapper.h

```diff
@@ -84,35 +98,73 @@ class PKGWrapper final : public IPackageWrapper
-                    std::getline(file, line))
-                {
-                    m_name = getValueFnc(line);
-                }
-                else if (line == "<key>CFBundleShortVersionString</key>" &&
-                         std::getline(file, line))
-                {
-                    m_version = getValueFnc(line);
-                }
-                else if (line == "<key>LSApplicationCategoryType</key>" &&
-                         std::getline(file, line))
-                {
-                    m_groups = getValueFnc(line);
-                }
-                else if (line == "<key>CFBundleIdentifier</key>" &&
-                         std::getline(file, line))
+                std::string line;
+                while(std::getline(data, line))
                 {
-                    m_description = getValueFnc(line);
+                    line = Utils::trim(line," \t");
+
+                    if (line == "<key>CFBundleName</key>" &&
+                        std::getline(data, line))
+                    {
+                        m_name = getValueFnc(line);
+                    }
+                    else if (line == "<key>CFBundleShortVersionString</key>" &&
+                            std::getline(data, line))
+                    {
+                        m_version = getValueFnc(line);
+                    }
+                    else if (line == "<key>LSApplicationCategoryType</key>" &&
+                            std::getline(data, line))
+                    {
+                        m_groups = getValueFnc(line);
+                    }
+                    else if (line == "<key>CFBundleIdentifier</key>" &&
+                            std::getline(data, line))
+                    {
+                        m_description = getValueFnc(line);
+                    }
                 }
             }
+        };
+
+        if (isBinary)
+        {
+            auto xmlContent { binaryToXML(filePath) };
+            getDataFnc(xmlContent);
         }
+        else
+        {
+            std::fstream file { filePath, std::ios_base::in };
+            if (file.is_open())
+            {
+                getDataFnc(file);
+            }
+        }
+    }
+
+    std::stringstream binaryToXML(const std::string& filePath)
+    {
+        plist_t rootNode { nullptr };
+        const auto binaryContent { Utils::getBinaryContent(filePath) };
+
+        // plist C++ APIs calls - to be used when Makefile and external are updated.
+        // const auto dataFromBin { PList::Structure::FromBin(binaryContent) };
+        // const auto xmlContent { dataFromBin->ToXml() };
+
+        // Content binary file to plist representation
+        plist_from_bin(&binaryContent[0], binaryContent.size(), &rootNode);
+        char* xml { nullptr };
+        uint32_t length { 0 };
+        // plist binary representation to XML
+        plist_to_xml(rootNode, &xml, &length);
+        std::string xmlContent{xml, xml+length};
```

Comment:
an xml != nullptr check should be added here to avoid a potential crash.

Reply:
Will do.